### ISSUE TITLE
Synchronize features with nextcloud server app

### ISSUE THREAD

**Issue by maenujem** [dmfs/opentasks#674]
It would be nice if the android app would have (only) the same features/fields visible for the user and in the same order as the server app in nextcloud. To be more flexible with the android app, there could also be a switch to activate (=hide fields/features not available in nextcloud) compatibility with nextcloud.

Right now the server app has the following fields:
* task list, 
* title, 
* description,
* start date,
* due date,
* all day,
* priority,
* percent completed, 
* (categories - do not work) -> use task lists instead, 
* subtasks (as markdown checkbox in description)

and the android app has the following fields:
* task list,
* title,
* description,
* start date,
* due date,
* all day,
* priority,
* percent complete,
* check list (subtasks),
* status,
* location,
* timezone,
* completed date,
* privacy,
* url

**Comment by avuton** [comment]
Yeah, I came here beause it appears subtasks are, unfortunately, implemented differently. I'm guessing this is probably the best bug to watch in hopes that this problem disappears eventually.

**Comment by godfuture** [comment]
Even there is a default alert created, custom alerts are missing in the Android apps. Relative and absolute ones.

**Comment by Ryonez** [comment]
Yup, thing breaks syncing with Nextcloud.

It gets the tasks, adds a timezone field, and then errors when it tries to sync back.

**Comment by 466** [comment]
> Yeah, I came here beause it appears subtasks are, unfortunately, implemented differently. I'm guessing this is probably the best bug to watch in hopes that this problem disappears eventually.

Same reason for me, though it's an owncloud instance the sync does not work propperly with.

[ owncloud instead of nextcloud - I guess the server implementations are the same(?) ]

**Comment by Ryonez** [comment]
@446 

> Same reason for me, though it's an owncloud instance the sync does not work propperly with.

Nextcloud forked of from Owncloud some time ago. Would make sense that they share some of the same bugs.

**Comment by dmfs** [comment]
Our major concern is support of RFC 5545, not a particular client/server application. We already have plans to allow configuration of visible fields and their display order.
Unfortunately, just like many other features this needs to wait until we have less resources bound in paid projects. Happy to accept PRs and answer questions regarding architecture & design, though.
